src/install.c

```diff
@@ -152,21 +152,6 @@ gboolean determine_slot_states(GError **error)
 		}
 	}
 
-	/* Determine active group members */
-	for (GList *l = slotlist; l != NULL; l = l->next) {
-		RaucSlot *s = g_hash_table_lookup(r_context()->config->slots, l->data);
-		g_assert_nonnull(s);
-
-		if (s == booted) {
-			s->state = ST_BOOTED;
-			g_debug("Found booted slot: %s on %s", s->name, s->device);
-		} else if (s->parent && s->parent == booted) {
-			s->state = ST_ACTIVE;
-		} else {
-			s->state = ST_INACTIVE;
-		}
-	}
-
 	if (!booted) {
 		if (g_strcmp0(r_context()->bootslot, "/dev/nfs") == 0) {
```

Comment:
Shouldn't the slots be marked as inactive in this case? Otherwise, we won't be able to install to them. We probably need to handle an external boot differently than a failure to determine slot states.

Reply:
Indeed, yes. We explicitly need to set them to 'invalid'. This also does not seem to be covered by any test, yet.

I'll update the PR accordingly.